lib/Jsrt/Jsrt.cpp

```diff
@@ -1157,6 +1157,30 @@ CHAKRA_API JsNumberToInt(_In_ JsValueRef value, _Out_ int *asInt)
     END_JSRT_NO_EXCEPTION
 }
 
+CHAKRA_API JsNumberToInt64(_In_ JsValueRef value, _Out_ int64_t *result)
+{
+    VALIDATE_JSREF(value);
+    PARAM_NOT_NULL(result);
+
+    BEGIN_JSRT_NO_EXCEPTION
+    {
+        if (Js::TaggedInt::Is(value))
+        {
+            *result = Js::TaggedInt::ToInt64(value);
+        }
+        else if (Js::JavascriptNumber::Is_NoTaggedIntCheck(value))
+        {
+            *result = Js::JavascriptConversion::ToInt64(Js::JavascriptNumber::GetValue(value));
```

Comment:
Does this case apply to `Number` objects, e.g. the result of `new Number(1)`? If so, this conversion can be re-entrant into script via a custom `valueOf` I think?

Reply:
No, this is explicitly checking for the tagged number case/JavascriptNumber case. Number objects are JavascriptNumberObjects. That being said, should we be checking for JavascriptNumberObject here? It looks like the existing implementations of JsNumberTo* don't handle Number objects, they likely require the caller to call JsConvertValueToNumber first.